### DISCUSSION TITLE
ModuleNotFoundError: No module named 'selenium'

### DISCUSSION THREAD

**Discussion by MikaTsur** [community/community#26329] [Actions]
<p>Hi,<br>
Im receiving:</p>
<p>ModuleNotFoundError: No module named ‘selenium’</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/4/a47f4e12338bc8ca04d0284cdd4b302e97044433.png" data-download-href="/uploads/short-url/ntd3oAuZZASkxIDMEig1kN6R7ZV.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/86562123/181096958-f25c94cc-b1a0-4ff7-bf23-5dd5ac9aba11.png" alt="image" data-base62-sha1="ntd3oAuZZASkxIDMEig1kN6R7ZV" width="629" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/4/a47f4e12338bc8ca04d0284cdd4b302e97044433_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">636×505 15.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>My ymel:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/3/934713841d68f6f165815a375ffa4b7d783d2595.png" data-download-href="/uploads/short-url/l0St25MccYkaeYwJGacDeLfOHid.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/86562123/181096959-c38c99a7-4762-4988-8fff-928b557044b1.png" alt="image" data-base62-sha1="l0St25MccYkaeYwJGacDeLfOHid" width="458" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/3/934713841d68f6f165815a375ffa4b7d783d2595_2_458x500.png, https://aws1.discourse-cdn.com/github/original/3X/9/3/934713841d68f6f165815a375ffa4b7d783d2595.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/3/934713841d68f6f165815a375ffa4b7d783d2595.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/3/934713841d68f6f165815a375ffa4b7d783d2595_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">652×711 22.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>requirements.txt contains selenium:<br>
)</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/e/1e8509ae02b42ab3799ec2ea0fe6b6552f764fe7.png" data-download-href="/uploads/short-url/4lZm5EDdKJxeoFUis7rO3G5OIsL.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/86562123/181096960-6ba9972a-f585-41c0-819d-5e14a9fa0035.png" alt="image" data-base62-sha1="4lZm5EDdKJxeoFUis7rO3G5OIsL" width="638" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/e/1e8509ae02b42ab3799ec2ea0fe6b6552f764fe7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">808×633 26.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>I will appreciate any help</p>

**Comment by airtower-luna** [comment]
<p>I’m pretty sure the <code>requirements.txt</code> gets never used. The command that would use it isn’t in a <code>run</code> step, it’s in an (unused) <code>run</code> parameter for the <code>actions/setup-python</code> step. You need to make the <code>pip install</code> a <code>run</code> step.</p>

**Comment by MikaTsur** [comment]
<p>Hi, Thanks for commenting!  sorry for being junior. I do not totally understand the problem.<br>
Please, can you be more specific and tell me what to add/move/ delete in which row?</p>
<p>I changed everything so many times and still didn’t understand.</p>

**Comment by MikaTsur** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/a/9af616bea0d179995bc560e8f8304f7e86780f44.png" data-download-href="/uploads/short-url/m6QLWPSoV24GP3jYragl319KD9W.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/86562123/181096961-1ff7aceb-1ead-40b5-a403-fc672494790c.png" alt="image" data-base62-sha1="m6QLWPSoV24GP3jYragl319KD9W" width="438" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/a/9af616bea0d179995bc560e8f8304f7e86780f44_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">523×597 14 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/6/26c6ced2505fafaacbce3a9da111d14af61ba07a.png" data-download-href="/uploads/short-url/5x25rCCX3q7Kl9Kr5wL4nCttgX8.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/86562123/181096962-59997d78-13fa-4fe5-b8a9-a6dcb0c26bbf.png" alt="image" data-base62-sha1="5x25rCCX3q7Kl9Kr5wL4nCttgX8" width="416" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/6/26c6ced2505fafaacbce3a9da111d14af61ba07a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">568×682 22.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by ViacheslavKudinov** [comment]
<p>Hi <a class="mention" href="/mikatsur">@MikaTsur</a><br>
What I see is you are mixing <code>actions/setup-python</code> with <code>run</code> as one step.<br>
I would suggest something eg:</p>
<pre><code class="lang-auto">steps:
  - uses: actions/checkout@v3
  - uses: actions/setup-python@v3
    with:
      python-version: '3.x'
      architecture: x64
  - run: |
      python -m pip install --upgrade pip
      pip install -r requirements.txt
  - run: Testing3.py
</code></pre>
<p>PS sorry if some indentation issues. Difficult to see it in form.</p>

**Comment by airtower-luna** [comment]
<p>Thank you <a class="mention" href="/viacheslavkudinov">@ViacheslavKudinov</a>, that’s exactly what I meant. :slightly_smiling_face:</p>
<p><a class="mention" href="/mikatsur">@MikaTsur</a> The important thing to understand here is YAML syntax. YAML can represent many kinds of data, not just Actions workflows. The <code>steps</code> element of a job is a list. Each step is an element in the list, each marked with a dash at the same indentation level. <code>run</code> needs to be right in that element to make it a <em>run step</em>, which runs a command. See also: <a href="https://www.yaml.info/learn/index.html" class="inline-onebox">Learn - yaml.info</a></p>

**Comment by MikaTsur** [comment]
<p>Thank you so much!!! finally I see a progress and the new error -<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/3/93a1dc93fc0a0e656d8e3f56d7a82ba087d97e2e.png" data-download-href="/uploads/short-url/l40YsiWjTSCaSOXklnmnU3miatg.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/86562123/181096963-21b10cb0-dcb9-4ac6-bb01-120b0661846f.png" alt="image" data-base62-sha1="l40YsiWjTSCaSOXklnmnU3miatg" width="690" height="416" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/3/93a1dc93fc0a0e656d8e3f56d7a82ba087d97e2e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1000×603 23.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>What should I do here?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You might need to explicitly call the Python interpreter to run the script:</p>
<pre><code class="lang-yaml">  - run: python Testing3.py
</code></pre>

**Comment by ViacheslavKudinov** [comment]
<p><a class="mention" href="/mikatsur">@MikaTsur</a><br>
I would recommend to start there <a href="https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python" class="inline-onebox" rel="noopener nofollow ugc">Building and testing Python - GitHub Docs</a><br>
also great to check other learnings <a href="https://docs.github.com/en/actions/guides" class="inline-onebox" rel="noopener nofollow ugc">Guides for GitHub Actions - GitHub Docs</a></p>

**Comment by MikaTsur** [comment]
<p>It is look much better  now!!! I see lines from python codes in the github logs.<br>
last error- :slightly_smiling_face:  ‘charmap’ codec can’t encode characters in position 0-2: character maps to </p>
<p><a href="https://github.com/MikaTsur/scedualAppointment/runs/6636251592?check_suite_focus=true#step:5:26" rel="noopener nofollow ugc">25</a></p>
<p>Something related to UTF-8 I think</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="MikaTsur" data-post="11" data-topic="252946">
<div class="title">
<div class="quote-controls"></div>
 MikaTsur:</div>
<blockquote>
<p>‘charmap’ codec can’t encode characters in position 0-2: character maps to</p>
</blockquote>
</aside>
<p>I can’t open the link, but that sounds like a problem with string processing in your code. Saying more would require seeing the code and full error message.</p>
